pkg/types/ccipocr3/common_types.go

```diff
@@ -7,6 +7,26 @@ import (
 	"strings"
 )
 
+// UnknownAddress represents a raw address with an unknown encoding.
+type UnknownAddress []byte
```

Comment:
Thinking out loud, but do we need some kind of equals/compare method then? How do we want to use it in the plugin code?

```go
func (u *UnknownAddress) Compare(other UnknownAddress, selector ChainSelector) int
```

Reply:
What if the unknown addresses are from two different families, would it ever make sense to compare them? I'm imagining this PR as the first step towards a final solution which has `Address` and `EncodedAddress` types which know how to convert between each other.

Maybe at that point they could be compared.

Some experiments from a couple weeks ago: https://github.com/smartcontractkit/chainlink-ccip/pull/158

Reply:
Yeah, good question, I think it would be great to start with the example usage. If `Message` stores all addresses as `UnknownAddress`, it might be very cumbersome for the customer of that interface to do the casting back and forth everytime they need to compare addresses to each other.

That being said, maybe Compare/Equals should return an `error` when trying to compare incompatible address types? On the other hand, Message has `SourceChainSelector` / `DestChainSelector` so we should be able to automatically convert these addresses to the proper types, no?

Reply:
I think it's already the case that working with `Bytes` is a little cumbersome. For comparison you would call the `String()` function.

This first change improves type safety and readability slightly by adding a specific type that tells you if it's encoded or not.

I'm imagining the CCIPReader would convert these to a the real address type (with validation) at which point you no longer have to worry about the chain selector or errors during encode/decode.